p8_integration_tests/quick_test/test_change_neuron_parameters_between_runs/test_initialize.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2017-2019 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import spynnaker8 as p
+from p8_integration_tests.base_test_case import BaseTestCase
+
+
+class TestInitialize(BaseTestCase):
+
+    # pop.set_inital_value is not a standard PyNN call, but
+    # we allow it on SpiNNaker, so we should test it (between runs)
+
+    def set_initialize_between_runs(self):
+        runtime1 = 5
+        runtime2 = 5
+        runtime3 = 5
+
+        p.setup(timestep=1.0)
+
+        pop = p.Population(3, p.IF_curr_exp())
+        pop.record(['v'])
+        self.assertEquals([-65, -65, -65], pop.initial_values["v"])
+
+        pop.initialize(v=-64)
+        self.assertEquals([-64, -64, -64], pop.initial_values["v"])
+        p.run(runtime1)
+
+        self.assertEquals([-64, -64, -64], pop.initial_values["v"])
+        pop.initialize(v=-62)
+        self.assertEquals([-62, -62, -62], pop.initial_values["v"])
+        p.run(runtime2)
+
+        self.assertEquals([-62, -62, -62], pop.initial_values["v"])
+        id_mixin = pop[1]
+        id_mixin.initialize(v=-60)
+        # v on not changed is now the current state not initital value
+        self.assertNotEqual(-60, pop.initial_values["v"][0])
+        self.assertNotEqual(-62, pop.initial_values["v"][0])
+        self.assertEquals(-60, pop.initial_values["v"][1])
+        self.assertNotEqual(-60, pop.initial_values["v"][2])
+        self.assertNotEqual(-62, pop.initial_values["v"][2])
+        p.run(runtime3)
+
+        p.reset()
+        neo = pop.get_data('v')
```

Comment:
this neo inst used. is there a reason for it being here? line 73 overrides it afterall

Reply:
if its used to set up initial state, shouldnt it be _ = pop_get_data('v')

Reply:
this neo inst used. = Agreed removed.

shouldnt it be _ : NO! get_data is the Standard PyNN call to get the value (of in this case v) for each timestep not just the initial state.